From 58568697a90844d98c7602902df8084297f19c1b Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Sun, 20 Dec 2009 13:42:21 +0000
Subject: Avoid requesting drag data if target is GDK_NONE (bug #5771).

Patch provided by k.blammo@gmail.com.

Ported to thunar 1.0.1 for Xubuntu/Debian.
---
diff --git a/thunar/thunar-standard-view.c b/thunar/thunar-standard-view.c
index 83139fc..8661861 100644
--- a/thunar/thunar-standard-view.c
+++ b/thunar/thunar-standard-view.c
@@ -2857,7 +2858,8 @@ thunar_standard_view_drag_motion (GtkWidget          *view,
       else
         {
           /* request the drag data from the source */
-          gtk_drag_get_data (view, context, target, time);
+          if (target != GDK_NONE)
+            gtk_drag_get_data (view, context, target, time);
         }
 
       /* tell Gdk whether we can drop here */
--
cgit v0.8.2.1
